test/tstQueryTreeManufacturedSolution.cpp

```diff
@@ -196,16 +202,18 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(structured_grid, TreeTypeTraits,
   offset_host = Kokkos::create_mirror_view(offset);
   Kokkos::deep_copy(offset_host, offset);
 
+  std::vector<int> ref_counts(ArborX::lastElement(offset_host));
+  std::vector<int> zeros(ref_counts.size());
+
   for (int i = 0; i < nx; ++i)
     for (int j = 0; j < ny; ++j)
       for (int k = 0; k < nz; ++k)
       {
         int index = ind(i, j, k);
         for (int l = offset_host(index); l < offset_host(index + 1); ++l)
-        {
-          BOOST_TEST(ref[index].count(indices_host(l)) != 0);
-        }
+          ref_counts[l] = ref[index].count(indices_host(l));
       }
+  BOOST_TEST(ref_counts != zeros, tt::per_element());
```

Comment:
also here

Reply:
See reasoning in [comment](https://github.com/arborx/ArborX/pull/471#discussion_r568873402).